src/main/java/com/mojang/brigadier/tree/CommandNode.java

```diff
@@ -42,30 +49,73 @@ protected CommandNode(final Command<S> command, final Predicate<S> requirement,
         this.forks = forks;
     }
 
+    /**
+     * Returns the command to execute when executing this command.
+     *
+     * @return the command to execute when executing this command or null if not set
+     */
     public Command<S> getCommand() {
         return command;
     }
 
+    /**
+     * Returns all child commands.
+     *
+     * @return all child commands
+     */
     public Collection<CommandNode<S>> getChildren() {
         return children.values();
     }
 
+    /**
+     * Returns a child with the given {@link #getName}.
+     *
+     * @param name the name of the child command
+     * @return the child with that name or null if not found
+     */
     public CommandNode<S> getChild(final String name) {
         return children.get(name);
     }
 
+    /**
+     * Returns the command node to redirect to.
+     *
+     * @return the command node to redirect to or null if none
+     */
     public CommandNode<S> getRedirect() {
         return redirect;
     }
 
+    /**
+     * Returns the redirect modifier to apply when redirecting.
+     *
+     * @return the redirect modifier to apply when redirecting or null if none
+     */
     public RedirectModifier<S> getRedirectModifier() {
         return modifier;
     }
 
+    /**
+     * Checks whether the given command source can use this command.
+     * <p>
+     * This just checks the {@link #getRequirement()} predicate, which could e.g. check for permissions.
+     *
+     * @param source the command source to check for
+     * @return true if the given command source can use this command
+     */
     public boolean canUse(final S source) {
         return requirement.test(source);
     }
 
+    /**
+     * Adds a new child node to this command node.
+     * <p>
+     * This will replace commands with the same name.
+     *
+     * <strong>You are not allowed to add children to commands with a {@link #getRedirect()} target!</strong>
```

Comment:
Ideally an exception should be thrown here, though not part of this pull request